dataset/bin.cpp

```diff
@@ -453,9 +453,32 @@ DataArray groupby_concat_bins(const DataArrayConstView &array,
       builder.groups(), {reductionDim});
 }
 
+namespace {
+void validate_bin_args(const std::vector<VariableConstView> &edges,
+                       const std::vector<VariableConstView> &groups) {
+  if (edges.empty()) {
+    if (groups.empty()) {
+      throw except::BucketError(
+          "Arguments 'edges' and 'groups' of scipp.bin are "
+          "both empty. At least one must be set.");
+    }
+  } else {
+    for (std::size_t i = 0; i < edges.size(); ++i) {
+      if (const auto &shape = edges[i].dims().shape();
+          std::any_of(shape.begin(), shape.end(),
+                      [](const scipp::index s) { return s == 1; })) {
+        throw except::BucketError("Not enough bin edges in dim " +
+                                  std::to_string(i) + ". Need at least 2.");
+      }
+    }
+  }
+}
```

Comment:
Question: here you do not check that `groups` has length > 1.
I guess that's fine as it is only `edges` that have to have length 2?

Reply:
I think so, providing length 0 groups should in principle create dimension with extent 0, which is allowed. I think having a test that this actually works would be useful though?